src/components/UserSelector.tsx

```diff
@@ -1,45 +1,41 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { clearPost } from '../features/Slices/selectedPostSlice';
+import { setAuthor } from '../features/Slices/author';
+import * as usersActions from '../features/Slices/usersSlice';
+export const UserSelector: React.FC = () => {
+  const [expanded, setExpanded] = useState(false);
+  const users = useAppSelector(state => state.users.users);
+  const author = useAppSelector(state => state.author.author);
+  const dispatch = useAppDispatch();
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
+  const selectAuthor = (user: User) => {
+    dispatch(setAuthor(user));
+    dispatch(clearPost());
+  };
 
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
-  const [expanded, setExpanded] = useState(false);
+  useEffect(() => {
+    if (users.length === 0) {
+      dispatch(usersActions.loadUsers());
+    }
+  }, [dispatch, users.length]);
```

Comment:
It's not a good practice to add array's length to the dependencies of useEffect. The reason is that the length of the array can change, but the array itself might still contain the exact same elements. It's better to add the whole array to the dependencies. In your case, you can omit users from the dependencies entirely, because you only want to fetch users when the component mounts.